File: app/src/main/java/nl/rsdt/japp/jotial/maps/wrapper/google/GooglePolygon.java
package nl.rsdt.japp.jotial.maps.wrapper.google;

import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.PolygonOptions;

import org.osmdroid.util.GeoPoint;
import org.osmdroid.views.MapView;

import java.lang.reflect.Array;
import java.util.ArrayList;

import nl.rsdt.japp.jotial.maps.wrapper.IPolygon;

/**
 * Created by mattijn on 08/08/17.
 */

public class GooglePolygon implements IPolygon{
    private final com.google.android.gms.maps.model.Polygon googlePolygon;

    public GooglePolygon(com.google.android.gms.maps.model.Polygon polygon) {
        googlePolygon = polygon;
    }

    @Override
    public void remove() {
        googlePolygon.remove();
    }

    @Override
    public void setStrokeWidth(int strokeWidth) {
        googlePolygon.setStrokeWidth(strokeWidth);
    }

    public void setFillColor(int color) {
        googlePolygon.setFillColor(color);
    }

    @Override
    public void setVisible(boolean visible) {
        googlePolygon.setVisible(visible);
    }
}

File: app/src/main/java/nl/rsdt/japp/jotial/maps/wrapper/google/GooglePolygon.kt
package nl.rsdt.japp.jotial.maps.wrapper.google

import nl.rsdt.japp.jotial.maps.wrapper.IPolygon

/**
 * Created by mattijn on 08/08/17.
 */

class GooglePolygon(private val googlePolygon: com.google.android.gms.maps.model.Polygon) : IPolygon {

    override fun remove() {
        googlePolygon.remove()
    }

    override fun setStrokeWidth(strokeWidth: Int) {
        googlePolygon.strokeWidth = strokeWidth.toFloat()
    }

    override fun setFillColor(color: Int) {
        googlePolygon.fillColor = color
    }

    override fun setVisible(visible: Boolean) {
        googlePolygon.isVisible = visible
    }
}
